components/script/dom/document.rs

```diff
@@ -2115,23 +2108,19 @@ impl Document {
         self.fullscreen_element.set(element);
     }
 
-    pub fn set_allow_fullscreen(&self, value: bool) {
-        self.allow_fullscreen.set(value);
-    }
-
     pub fn get_allow_fullscreen(&self) -> bool {
         // https://html.spec.whatwg.org/multipage/#allowed-to-use
         match self.browsing_context() {
             // Step 1
-            None => true,
+            None => false,
             Some(_) => {
                 // Step 2
                 let window = self.window();
                 if window.is_top_level() {
                     true
                 } else {
                     // Step 3
-                    self.allow_fullscreen.get()
+                    window.GetFrameElement().map(|el| el.has_attribute(&local_name!("allowfullscreen"))).unwrap_or(false)
```

Comment:
Turns out we can use `map_or(false, ...)` instead of `map(...).unwrap_or(false)`.